### ISSUE TITLE
Codelens URL JAX-RS slows down the Java editor

### ISSUE THREAD

**Issue by angelozerr** [redhat-developer/intellij-quarkus#790]
When Java class is annotated with @Path in class, the Java editor is very slow. After some investigation, the problem comes from the URL codelens with https://github.com/redhat-developer/intellij-quarkus/blob/02d2869c4ebc89ebead5863fe2b3a372c2c28b3d/src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/internal/jaxrs/java/JaxRsCodeLensParticipant.java#L139 which checks if server is available. If you update the method by returning true, th eperformance are very good.

To reproduce the problem create a Foo class like this:

```java
package org.acme.rest.client;

import javax.ws.rs.Path;

@Path("")
public class Foo {

    
}

When you comment @Path annotation, the codelens URL is not computed and it is very fast, if you have @Path annotation it is very slow.

See the following demo:

```
![CodeLensURLIJSlow](https://user-images.githubusercontent.com/1932211/231397161-19b91c54-6853-4054-b149-52b8b16cdd2d.gif)

@jeffmaury have you some idea how we could fix this issue? The easy fix is to remove this check of server available, but perhaps a better fix should be done in https://github.com/redhat-developer/intellij-quarkus/blob/02d2869c4ebc89ebead5863fe2b3a372c2c28b3d/src/main/java/com/redhat/devtools/intellij/quarkus/lsp/QuarkusLanguageClient.java#L147 by improving runAsBackground?

**Comment by angelozerr** [comment]
> does lsp4ij cancel/debounce codelens requests on type?

I don't think so,I have create the issue https://github.com/redhat-developer/intellij-quarkus/issues/791

**Comment by angelozerr** [comment]
I discovered that a user complain about this problem in https://github.com/redhat-developer/intellij-quarkus/issues/487

**Comment by jeffmaury** [comment]
Need to check if the code lens is computed in the UI thread or not

**Comment by angelozerr** [comment]
@jeffmaury I think I have understood the problem. It is because IJ InlayHintsProvider that we use doesn't support asynchronize CompletableFuture like Eclipse IDE and vscode supports.

It is the reason why in Eclipse IDE and vscode we have not this problem. The check of server available is done on background and even if it takes some times, the editor is not frozen.

In IJ, InlayHintsProvider doesn't support ComletableFuture. It is perhaps we use intellij community and perhaps ultimate provides CompletableFuture for Inlay hint?

**Comment by angelozerr** [comment]
I fear that there is no solution to fix this issue, when I set a breakpoint at https://github.com/redhat-developer/intellij-quarkus/blob/be26eb2aa0ccd4734991e01ff6b40b009c42abec/src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/operations/codelens/LSPCodelensInlayProvider.java#L61 the IJ IDE is frozen.

I would prefer displaying codelens even if server is not available instead of have some freeze with Java editor.
